presto-kafka/src/main/java/io/prestosql/plugin/kafka/PlainTextKafkaProducerFactory.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.kafka;
+
+import com.google.common.collect.ImmutableSet;
+import io.prestosql.spi.HostAddress;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.common.serialization.ByteArraySerializer;
+
+import javax.inject.Inject;
+
+import java.util.Properties;
+import java.util.Set;
+
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.joining;
+import static org.apache.kafka.clients.producer.ProducerConfig.ACKS_CONFIG;
+import static org.apache.kafka.clients.producer.ProducerConfig.BOOTSTRAP_SERVERS_CONFIG;
+import static org.apache.kafka.clients.producer.ProducerConfig.LINGER_MS_CONFIG;
+
+public class PlainTextKafkaProducerFactory
+{
+    private final Properties properties;
+
+    @Inject
+    public PlainTextKafkaProducerFactory(KafkaConfig kafkaConfig)
+    {
+        requireNonNull(kafkaConfig, "kafkaConfig is null");
+        Set<HostAddress> nodes = ImmutableSet.copyOf(kafkaConfig.getNodes());
+        this.properties = new Properties();
+        this.properties.put(
+                BOOTSTRAP_SERVERS_CONFIG,
+                nodes.stream()
+                        .map(HostAddress::toString)
+                        .collect(joining(",")));
+        this.properties.put(ACKS_CONFIG, "all");
+        this.properties.put(LINGER_MS_CONFIG, 5);  // reduces number of requests sent, adds 5ms of latency in the absence of load
+    }
+
+    public KafkaProducer<byte[], byte[]> create()
+    {
+        return new KafkaProducer<>(properties, new ByteArraySerializer(), new ByteArraySerializer());
```

Comment:
`properties` is mutable, so defensive copy would be needed here

- change `Properties properties` field to `Map<...>  properties` and initialize it using `ImmutableMap`
- by using `Map<String, Object>` you can call the overload of `KafkaProducer` that does not require you to convert to `Properties`